### PR TITLE
fix: disconnected callback guards

### LINKED ISSUE #666: bug: Tests fails after version > 4.6.0
### Prerequisites

- [X] I know [how to contribute a proper bug report](https://github.com/emdgroup-liquid/liquid/blob/main/CONTRIBUTING.md#reporting-bugs).
- [X] I agree to follow the [Code of Conduct](https://github.com/emdgroup-liquid/liquid/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/emdgroup-liquid/liquid/issues?q=is%3Aopen+is%3Aissue+label%3Abug) that already report this problem, without success.

### Liquid version

> 4.6.0

### Framework bindings

React

### Current behavior

We were using Liquid version 4.13.0 but we wanted to upgrade to newest version. We found out that 4.6.0 introduces some problems to our Jest + RTL tests. We are using CRA and we added the liquid component library into the `transformIgnorePatterns` to notify Jest that we want to compile Liquid library to CommonJS and so on.

### Expected behavior

Tests should be working as expected in versions above 4.5.0.

### Steps to reproduce

1. Use CRA
2. Add "node_modules/(?!@emdgroup-liquid)/" to transformIgnorePatterns for notify Jest that it should transform ESM modules
3. Use LdModal in component
4. Render the component with RTL

### Code reproduction URL

https://codesandbox.io/p/github/emdgroup-liquid/liquid-sandbox-cra-tailwind/csb-lqyi2w/draft/fervent-curran?file=%2Fsrc%2FApp.tsx&selection=%5B%7B%22endColumn%22%3A52%2C%22endLineNumber%22%3A8%2C%22startColumn%22%3A52%2C%22startLineNumber%22%3A8%7D%5D

### Additional information

Additional case with modal component, tests are not failing but weird error message popped out: https://codesandbox.io/p/github/emdgroup-liquid/liquid-sandbox-cra-tailwind/csb-585zse/draft/great-meitner?file=%2Fsrc%2FApp.tsx&selection=%5B%7B%22endColumn%22%3A33%2C%22endLineNumber%22%3A5%2C%22startColumn%22%3A33%2C%22startLineNumber%22%3A5%7D%5D
We also tried to use experimental ESM support for Jest and it seems that it's working then. But the problem is we have lots of tests already written but this experimental ESM support is a braking change so it will be really hard to switch to that.

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

This PR resolves issues where non-hydrated components throw errors when run in unit tests by adding guards.

Resolves #666

## Type of change

Please delete options that are not relevant.

- [x] Bugfix

## Is it a breaking change?

- [x] No

# How Has This Been Tested?

- [x] unit tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: src/liquid/components/ld-stepper/ld-step/test/ld-step.spec.tsx [hunk 1]
```diff
@@ -306,4 +322,14 @@ describe('ld-step', () => {
 
     expect(button.focus).toHaveBeenCalled()
   })
+
+  it('does not throw when setting inner focus on nothing', async () => {
+    const component = new LdStep()
+    await component.focusInner()
+  })
```
Comment (renet) [inline src/liquid/components/ld-stepper/ld-step/test/ld-step.spec.tsx:329]: Is this really worth testing, as the component is not meant to be used this way and does not work when being used like that anyway?

**Review by renet — CHANGES_REQUESTED** [review]

File: src/liquid/components/ld-button/ld-button.tsx [hunk 2]
```diff
@@ -169,10 +165,13 @@ export class LdButton implements InnerFocusable, ClonesAttributes {
     button.remove()
   }
 
-  private handleClick = (ev: MouseEvent) => {
-    const ariaDisabled = this.button.getAttribute('aria-disabled')
+  private isAriaDisabled = () => {
+    const ariaDisabledAttr = this.el.getAttribute('aria-disabled')
+    return ariaDisabledAttr && ariaDisabledAttr !== 'false'
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-button/ld-button.tsx:170]: This is on my list, actually for today. I noticed that we have issues with `aria-disabled` all over the place and I want to fix it in the next PR. The thing is: `aria-disabled` can have `yolo` as its value, which is interpreted as `true`. Only `'false'`, `''` and `undefined` are interpreted as `false`...
**TLDR**: This is not wrong and will be unified in the next PR.

File: src/liquid/components/ld-button/ld-button.tsx [hunk 3]
```diff
@@ -143,7 +138,8 @@ export class LdButton implements InnerFocusable, ClonesAttributes {
     this.el.removeEventListener('click', this.handleClick, {
       capture: true,
     })
-    this.attributesObserver?.disconnect()
+    /* istanbul ignore if */
+    if (this.attributesObserver) this.attributesObserver.disconnect()
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-button/ld-button.tsx:142]: No, this is not what I'm doing here. The `if` part is new, and actually fixes the issue described in the PR / the referenced issue on GitHub.

For other parts in this PR I may have changed optional chaining to explicit if statements in order for istanbul to ignore a specific branch, which I think is acceptable, because I didn't change any logic there. But here the change in logic is intentional and has nothing to do with test coverage.

File: src/liquid/components/ld-input/ld-input.tsx [hunk 4]
```diff
@@ -357,8 +357,8 @@ export class LdInput implements InnerFocusable, ClonesAttributes {
   }
 
   disconnectedCallback() {
-    /* istanbul ignore next */
-    this.attributesObserver?.disconnect()
+    /* istanbul ignore if */
+    if (this.attributesObserver) this.attributesObserver.disconnect()
```
[the remark this answers is not in the document]
Comment (borisdiakur) [reply src/liquid/components/ld-input/ld-input.tsx:361]: This is explicitly telling that we intentionally do not test this branch (because it's not worth it). This is better than seeing missing coverage and asking yourself over and over again whether we need to write tests. I'd like to keep it like this.

**Review by renet — APPROVED** [review]
